Aesthetic pack

pi-squared-studio opened the pull request:
# Description
Several aesthetic improvements for the top and bottom surfaces. Generally, they are designed for center-based Archimedes Chords and Octagram Spiral patterns. For other patterns, the action needs to be checked or resolved in algorithms.

## Anisotropic surfaces
Sometimes it turns out that the direction of infill changes the aesthetic appearance of the surface. This is clearly noticeable if you use multicolor or silk plastics. This option disables drawing surfaces in different directions, so they look noticeably better. At the same time, the printing speed is loosing.
<img width="1050" height="531" alt="image" src="https://github.com/user-attachments/assets/a57db7c8-b26f-4803-8628-d25d46922b4d" />
Some notes to ensure the cleanliness of the top surface:
- Set the filling density for the top and bottom surfaces to 75-90% to obtain a uniform layer without overflow on adjacent lines. Also, keep the patchwork_infill_flow_ratio value less than number one so that the tracks don't overlap when printing. 
- Set more hotend temperature to add shine. Try, for example, set the 240°C Instead of 220°C recommended by the manufacturer for PLA plastic. Make sure that the model is printed without problems and unnecessary smudges.
- Use the brand new and quality nozzles.
- Set a low surface printing speed. Usually 20-30 mm/s. Choose according to the best result.
- Adjust the Pressure Advance
- Apply Z-hop at least to the top surface.
- Reduce the retracts. Too frequent pressure drops make melted filament mass less uniform.
- Don't use classic wipe method. That's may dirty top surface. If the extends of pathlines out the pattern beyond the contour of the model, then you can specify a negative value to wipe_distance parameter so that all the stretching snot threads freeze outside the model.
- Use dry plastic. Do not allow it to leak out of the nozzle when traveling.
- Increase the speed of movement and acceleration of the hotend during travel to the maximum possible. This will reduce the amount of snotty plastic. Make sure that the kinematics of the printer can handle such loads.
- Ensure that the top surface for PLA plastic is cooled as much as possible.


## Center of surface patterns

<img width="369" height="121" alt="image" src="https://github.com/user-attachments/assets/aab85b90-cbc8-42f6-85d0-c9b1b25c865d" />

You can choose one of the 3 options for centering the axial pattern:
- for each surface (classic mode)
- for each model
- for each assembly

If the model has one body but several surfaces, then you can set both separate centering for each surface and set a global center for the whole model. The same parameter is available for assembly if the composition consists of several unrelated models.

<img width="1347" height="577" alt="image" src="https://github.com/user-attachments/assets/01405e8d-2c32-41b7-a886-b071b8b49de2" />

<img width="1347" height="577" alt="image" src="https://github.com/user-attachments/assets/33fd7079-4417-4d5e-a162-9b57e8ab06ff" />

## Precision surfaces

<img width="369" height="36" alt="image" src="https://github.com/user-attachments/assets/0299d9d5-1acc-4a75-9dd2-3430a9ca5bb0" />

This mod sets the precision overlay of patterns on the top and bottom surfaces, and partially fills the pores when using Fuzzy Skin. Both works with Classic wall generator mode and also with Arachne.

<img width="1347" height="683" alt="image" src="https://github.com/user-attachments/assets/dcfda640-0372-483b-90d7-3468cde7e187" />
<img width="1347" height="683" alt="image" src="https://github.com/user-attachments/assets/71e3953d-93ab-4d1f-bca3-b8b2e6c69a65" />
<img width="881" height="756" alt="image" src="https://github.com/user-attachments/assets/860ef2c2-3d02-4cf4-85d8-ee85a7d96edf" />
<img width="881" height="756" alt="image" src="https://github.com/user-attachments/assets/186e216f-4ae6-4870-87fc-207eb2708f4e" />

## Patchwork surfaces

<img width="369" height="372" alt="image" src="https://github.com/user-attachments/assets/2c171221-68d4-4199-986a-bfde39c68ef6" />

What are the options in this add-on:


- **Patchwork surfaces**: determines on which external surfaces the patchwork drawing will be applied
<img width="417" height="210" alt="image" src="https://github.com/user-attachments/assets/fcbe53f0-89a6-4630-84f9-87fee2c0e2e1" />


- **Patchwork direction** and **Alternate Tiles Direction**:
The angle of inclination of the masonry itself and the metalanguage element for setting the absolute/relative value of the tilt of the tiles. A value of '+0' or '360' causes a random tilt.
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/041c3544-e81f-4e75-b4eb-c875f388ed4b" />
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/f41f62e4-56f1-4d0f-a265-46eac3723c30" />
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/3cb336f7-4ae8-49a2-9efe-2eff978d8947" />
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/63dd8c48-2dae-4338-a8dc-e448e600889a" />

- height and width of the tile - and so everything is clear

- Vertical and horizontal joints:
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/2d9fb80e-07d6-464e-b595-644dbf74936a" />
negative values define an unfilled seam
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/f4b5c226-9ca1-4518-87b9-b1e76b523e5a" />


- **Patchwork centering** and **Subway tiling**:
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/6fa2a6ba-7a3d-42c2-a2fd-edea44ff6a94" />
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/acefd2ee-b109-43d7-b3c8-6321c2ac70d0" />
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/8b934f02-67ec-43e2-8f4f-cbc3b824bb08" />


- **Center of surface patterns**:  (for polar patterns)
When the Patchwork surfaces option is enabled, this parameter will work like this:
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/a369551e-86b5-4dc9-8fcd-08d97c7d71b8" />
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/41ed5a6e-54d9-4d01-aeac-07cebea81d7f" />
<img width="959" height="616" alt="image" src="https://github.com/user-attachments/assets/f179693b-1d16-453d-9feb-6d2290586a6a" />


#### Examples:

<img width="1440" height="614" alt="image" src="https://github.com/user-attachments/assets/ee080edf-71d1-4ad7-a889-0efc2f1d7fc3" />

## Also solved problems
+ an add-on that can extend the wipe path beyond the contour when entering a negative value in wipe_distance. It is very useful for the purity of the upper surface pattern.

+ precision infill/wall overlap for the classic perimeter generator. Previously, there was a bug.

+ remove some bugs

+ add fractional numbering in the infill rotation templates (that always decrypted in wiki)
<img width="1422" height="614" alt="image" src="https://github.com/user-attachments/assets/b892ff07-90ac-4135-8283-d0070f045ff6" />

+ grouping and sorting of surface infill elements in menu by it's dominant
<img width="400" height="482" alt="image" src="https://github.com/user-attachments/assets/23badcb8-bd64-4d26-8be5-36d25efcfeac" />

Also made for sparse infill. It seems that it looks more readable this way.
<img width="400" height="482" alt="image" src="https://github.com/user-attachments/assets/4d3d5a59-be8e-4ad6-93ee-5349f62f8728" />

pi-squared-studio committed:
Aesthetic Surfaces

pi-squared-studio committed:
Some updates

pi-squared-studio commented:
I checked the code, and rolled back a bit in some of the fixed features. As it turned out, the infill polylines are optimized several times in Orca, so that the actions of the primary optimization functions simply turn out to be unnecessary, and even harmful in terms of execution speed. Now the code is working as intended, but I can't provide the tests yet due to other difficulties related to retracts and snotty plastic from the nozzle...

pi-squared-studio commented:
Here are the first tests. Do not pay attention to the snot that inevitably appears with this method, but you can deal with them with the print settings. There's not much time right now to take a lot of beautiful photos. I am only describing how this method works.

<img width="780" height="679" alt="image" src="https://github.com/user-attachments/assets/93579a3b-710c-4f00-bbcd-accdfc7ad8a5" />

I took a tricolor silk PLA. All photos were taken from the top surface. The first one (upper left) is made without aesthetic surfaces. As you can see, the interlaced structure of the corners of the pattern is visible, where the printhead changes the printing direction. 
The second photo is the first implementation of this algorithm. Here, the printing direction is aligned, but the print segments are optimized.
The lower test is an attempt to work with retracts.  I replaced the printing nozzle. The temperature of the plastic is 220°C (against 240 in the first tests, which gives a more glossy appearance). The fill density is 80%, which makes it appear more granular. The printing direction is from the center to the edges, and strictly translational. As you can see, there are much fewer visual artifacts.

pi-squared-studio commented:
Another test of the modified algorithm. I used a 3-color (crimson-brick-turquoise) silk PLA.
<img width="631" height="608" alt="image" src="https://github.com/user-attachments/assets/c7a83089-d376-4fc0-87d5-a9e94663ac52" />

The bottom surface (Archimedes) turns out to be perfect. On the top (Octogram), surface defects are visible, They are obtained due to snots from the nozzle during large travel, and wet PLA which I used. (I haven't got a filament dryer at the moment).

pi-squared-studio committed:
Update algorythm

pi-squared-studio committed:
Align center of patterns to

pi-squared-studio committed:
Precision surfaces

SoftFever committed:
Merge branch 'main' into Aesthetic-top-and-bottom-surface-patterns

SoftFever committed:
Merge branch 'main' into Aesthetic-top-and-bottom-surface-patterns

SoftFever commented on src/libslic3r/ShortestPath.hpp:
```diff
@@ -26,8 +26,9 @@ std::vector<std::pair<size_t, bool>> chain_extrusion_paths(std::vector<Extrusion
 void                                 chain_and_reorder_extrusion_paths(std::vector<ExtrusionPath> &extrusion_paths, const Point *start_near = nullptr);
 
-Polylines 							 chain_polylines(Polylines &&src, const Point *start_near = nullptr);
+Polylines                            chain_polylines(Polylines&& src, const Point* start_near = nullptr);
```
Could you undo these format changes to keep the git history lean?

SoftFever commented on src/libslic3r/PrintConfig.cpp:
```diff
@@ -525,7 +536,7 @@ void PrintConfigDef::init_common_params()
     def->category = L("Quality");
     def->tooltip = L("Shrinks the initial layer on build plate to compensate for elephant foot effect.");
-    def->sidetext = "mm";	// milimeters, don't need translation
+    def->sidetext = L("mm");	// unlike signs, units are translated into some languages
```
Can we keep the PR focused on the relevant changes only? These unrelated changes make the code review extremely difficult.

SoftFever commented on src/libslic3r/PrintConfig.cpp:
```diff
@@ -2805,6 +2816,42 @@ void PrintConfigDef::init_fff_params()
+    def->mode    = comAdvanced;
+    def->set_default_value(new ConfigOptionBool(false));
+
+    def           = this->add("align_center_of_patterns", coEnum);
```
can we name this "center_of_infill_pattern"?

pi-squared-studio replied:
OK. It was initially difficult for me to give a name for this function.

SoftFever replied:
Internally, they are all infill patterns.  
For the display string (the label), feel free to call it surface.

SoftFever reviewed (changes requested):
Some quick thoughts:  
I like the "aesthetic_surfaces" and "align_center_of_patterns" ideas; I think they are useful.  
I have questions about the other changes; please review my code comments.  

I would like to suggest:  
1. Include changes for these two features only in this PR.  
2. Display these two options in the same place as other infill options in the UI.  
3. Enable or display these options only for center-based Archimedes Chords and Octagram Spiral patterns.

pi-squared-studio commented on src/libslic3r/ShortestPath.cpp:
```diff
@@ -998,6 +998,13 @@ std::vector<std::pair<size_t, bool>> chain_segments_greedy2(SegmentEndPointFunc
 	return chain_segments_greedy_constrained_reversals2_<PointType, SegmentEndPointFunc, false, decltype(could_reverse_func)>(end_point_func, could_reverse_func, num_segments, start_near);
 }
 
```
Hi! 
Whatever you deem necessary to do is within your rights!
At the moment, I'm still working on this project, and I wanted to look at the code again at the end and do the final cleaning of all kinds of garbage. Also make sure that some algorithms work correctly... But if You decide to fix the job at this stage, I can transfer it to Draft. And already make new functions in the new PR.

SoftFever replied:
I see.
Yes, it would be helpful to mark it as a draft or add [NOT READY] in the PR title if it isn't ready for review, so I can hold on the review.

pi-squared-studio commented on src/libslic3r/PrintConfig.cpp:
```diff
@@ -2805,6 +2816,42 @@ void PrintConfigDef::init_fff_params()
     def->mode = comAdvanced;
     def->set_default_value(new ConfigOptionInts{ -1 });
 
+    def           = this->add("aesthetic_surfaces", coBool);
```
Of course you can! But, I think that the addition to this function will be much more than the line drawing order.
Then I thought that when choosing this option, it would be possible to change others so that the effect would be the most advantageous. For example, the surface density... But, again, the final decision is yours.

SoftFever replied:
Could you please elaborate on it? In the current implementation, enabling this option will optimize line directions so that they are anisotropic. Did I miss anything?  
Another issue is that calling it "Aesthetic Surfaces" is very ambiguous; users won't have any idea what specific changes and impacts occur when the option is turned on or off.

pi-squared-studio commented:
> I would like to suggest:
OK, then I'll fix everything possible here, and transfer the new idea to the new PR.
I will give a small announcement - by the way, it will be called "Patchwork surfaces" )))
It's going to be a really cool thing... But now I'm faced with the problem that I can't properly separate the surfaces using a Clipper API. Then I'll postpone this project for a week...

SoftFever commented on src/libslic3r/PrintConfig.cpp:
```diff
@@ -2805,6 +2816,42 @@ void PrintConfigDef::init_fff_params()
+    def->mode     = comAdvanced;
+    def->set_default_value(new ConfigOptionEnum<AlignCenterOfPatterns>(AlignCenterOfPatterns::Each_Surface));
+
+    def           = this->add("precision_surfaces", coBool);
```
I see. Yes, it makes sense.  
I would suggest creating a separate PR for this improvement.

pi-squared-studio replied:
Initially, I thought they would be linked into one pack... I will do the following projects.
Give me 1-2 days to complete this PR. I just got really involved in another project, I didn't want to interrupt it.

pi-squared-studio commented:
@SoftFever 
After all, I decided to leave one PR for now. Since all the procedures are tied to each other and it is not possible to divide this branch into several different ones. After optimizing and completing the project, it may be easier to do this.

Here's a little teaser of what I'm doing at the moment: ))
<img width="1054" height="919" alt="image" src="https://github.com/user-attachments/assets/154609a0-5506-45a4-a42d-77df24acd66c" />

@ianalexis when I finish, will it be possible to contact you to correct the possible Wiki and some UI before adding all this to the code?

ianalexis commented:
> @ianalexis when I finish, will it be possible to contact you to correct the possible Wiki and some UI before adding all this to the code?

sure!
Im at discord, telegram, whatsapp, linkedin or comment in this new pr #10369

pi-squared-studio commented:
First test of patchwork! (not anisotropic yet)
<img width="635" height="475" alt="image" src="https://github.com/user-attachments/assets/03d9f33e-16f1-45eb-aebb-9a60323743a7" />
<img width="635" height="475" alt="image" src="https://github.com/user-attachments/assets/a7bd966e-67c8-49fb-a965-12e13ea2008a" />

pi-squared-studio commented:
I'll post the code soon. For now, for testing and improvements.

<img width="1293" height="646" alt="image" src="https://github.com/user-attachments/assets/95c1a1f5-9875-4297-9689-927e18011788" />
<img width="1293" height="646" alt="image" src="https://github.com/user-attachments/assets/7ff4731a-afd2-404a-9a09-23ca681a74c0" />
<img width="1293" height="646" alt="image" src="https://github.com/user-attachments/assets/1545cadb-c1d5-400f-ba12-625be49afa91" />
<img width="1293" height="646" alt="image" src="https://github.com/user-attachments/assets/650d6a6e-f2db-41ae-99ce-71f24f1192bd" />

There are still some problems with correctly displaying patterns from different angles.
The problem of displaying non-polar patterns when selecting the centering option for the assembly has not yet been solved. We need to find a procedure to trim (Thick)Polylines with other ThickPolylines.

pi-squared-studio committed:
Patchwork Surfaces

pi-squared-studio committed:
Merge branch 'Aesthetic-top-and-bottom-surface-patterns' of https://github.com/pi-squared-studio/OrcaSlicer into Aesthetic-top-and-bottom-surface-patterns

pi-squared-studio committed:
Update ShortestPath.hpp

pi-squared-studio committed:
Update ShortestPath.hpp

pi-squared-studio committed:
spell check

pi-squared-studio committed:
Merge branch 'main' into Aesthetic-top-and-bottom-surface-patterns

pi-squared-studio commented:
> Why should we leave the option to translate mm, which is the “universal symbol” for millimeters?

So I opened up... I forgot to translate that post into English.)))
It's just that in my language, these values have been translated for 200 years as. 
mm - мм
cm - см
m/s - м/с
this is not critical for me personally, I use the original language version of the Orca, but if it would be more convenient for the end user, it would be nice.

pi-squared-studio committed:
Rollback translate a units + find some bugs

pi-squared-studio committed:
Merge branch 'Aesthetic-top-and-bottom-surface-patterns' of https://github.com/pi-squared-studio/OrcaSlicer into Aesthetic-top-and-bottom-surface-patterns

ianalexis commented:
> So I opened up... I forgot to translate that post into English.)))

No te preocupes camarada! XD

> m/s - м/с

This surprised me, so I did some research and realized that you have the Russian designation/русское обозначение.  
Even so, I think we should keep the international/международное (Интернационал ?), although if you think that in the territory where Cyrillic (Кириллица) is used, the vast majority use the русское обозначение, we should make that change and perhaps have two versions of Русский, one with each version.

pi-squared-studio committed:
Finish code

pi-squared-studio committed:
remove own bugs

pi-squared-studio commented:
@SoftFever  @ianalexis 
Hi! I finish this code.
Look, I've tested it many times, and it works quite well for me. The main problem was that it included many parameters that influenced each other. I marked all critical comments //PPS: - in the final version, you can do whatever you want with the code, these comments can be deleted altogether if they are clear to understand, or contain incomprehensible interpretations. It is possible that this code was integrated into the wrong places of the code, so you can convert it to the desired criteria.
I touched on the some fundamental elements of the code a little, but this happened as an improvement in conditions when new functions were added and errors became noticeable. If you have any questions, ask me, I will answer for them in detail.

There are some minor unresolved issues.
- enabling the Precision Surfaces option does not always work when the G-code is generated. You need to switch any other option for this function to work.
- When I do an extreme Fuzzy Skin with Precision Surfaces, instead of one surface, several are formed, mainly the pores between the loops. And they are not always defined as Internal. There is no other way to do it, since the existing principle of forming walls and pouring. First, the infill areas are determined, then only the perimeters are formed, which are modified by Fuzzy Skin. And accordingly, the filling area also changes... Therefore, I had to write illogical code to sort such planes into internal and external based on preliminary data.

So far, this is just code with an inaccurate description in the UI for those functions. I am not a native English speaker, and I may have given them incorrectly. Need to give them a more accurate description.

Also ask me for the basis for the wiki. It would be better if you were given to me a template with theses that need to be filled with the material.

pi-squared-studio committed:
Merge branch 'main' into Aesthetic-top-and-bottom-surface-patterns

ianalexis commented on src/slic3r/GUI/Tab.cpp:
```diff
@@ -2454,15 +2466,15 @@ optgroup->append_single_option_line("skirt_loops", "others_settings_skirt#loops"
-        optgroup->append_single_option_line("fuzzy_skin_octaves", "others_settings_fuzzy_skin#skin-noise-octaves");
-        optgroup->append_single_option_line("fuzzy_skin_persistence", "others_settings_fuzzy_skin#skin-noise-persistence");
-        optgroup->append_single_option_line("fuzzy_skin_first_layer", "others_settings_fuzzy_skin#apply-fuzzy-skin-to-first-layer");
+        optgroup->append_single_option_line("fuzzy_skin", "others_settings_special_mode#fuzzy-skin");
+        optgroup->append_single_option_line("fuzzy_skin_mode", "others_settings_special_mode#fuzzy-skin-generator-mode");
+        optgroup->append_single_option_line("fuzzy_skin_noise_type", "others_settings_special_mode#fuzzy-skin-mode");
+        optgroup->append_single_option_line("fuzzy_skin_point_distance", "others_settings_special_mode#point-distance");
+        optgroup->append_single_option_line("fuzzy_skin_thickness", "others_settings_special_mode#skin-thickness");
+        optgroup->append_single_option_line("fuzzy_skin_scale", "others_settings_special_mode#skin-feature-size");
+        optgroup->append_single_option_line("fuzzy_skin_octaves", "others_settings_special_mode#skin-noise-octaves");
+        optgroup->append_single_option_line("fuzzy_skin_persistence", "others_settings_special_mode#skin-noise-persistence");
+        optgroup->append_single_option_line("fuzzy_skin_first_layer", "others_settings_special_mode#apply-fuzzy-skin-to-first-layer");
```
You have to revert this changes, it's fine as it is now.
I think you still have the change from some other old modification.

pi-squared-studio committed:
Updates

ianalexis commented:
About the error that "Validate Documentation" is giving to you.
They are becouse you didint add the fragment in the documentation.
You can do 1 of theese 3 options:
1. Remove the linkin (the second string) like
  ```c++
  optgroup->append_single_option_line("patchwork_tiles_alternate_direction");
  ```
2. Remove the fragment like
  ```c++
  optgroup->append_single_option_line("patchwork_tiles_alternate_direction", "strength_settings_top_bottom_shells");
  ```
3. Create the fragment in the doc and write something basic or WIP...

pi-squared-studio committed:
Update Tab.cpp

pi-squared-studio commented:
> You can do 1 of theese 3 options:

Ok, I just deleted the links to the documentation. The fact is that I am not sure about the correctness of the names in English of the labels, as well as their descriptions. I think we need to define them first, and then build a wiki based on these changes.

<img width="421" height="586" alt="image" src="https://github.com/user-attachments/assets/95d44316-d7e1-4781-bacc-bc75c6adf44e" />
<img width="421" height="586" alt="image" src="https://github.com/user-attachments/assets/3ca5dbcc-dd30-4e9a-8970-f537a0985500" />
<img width="481" height="586" alt="image" src="https://github.com/user-attachments/assets/d5f300d2-bcde-4644-a45e-800bbe90234b" />
<img width="481" height="586" alt="image" src="https://github.com/user-attachments/assets/1d25b8e7-3ccd-4ca0-ac95-e7a0865bdcbc" />

pi-squared-studio committed:
Merge branch 'main' into Aesthetic-top-and-bottom-surface-patterns

pi-squared-studio committed:
Some adds and usabilities

+ step settings into patchwork parameters
+ check determination for top_infill in precision surfaces
+ correct determination patchwork surface

pi-squared-studio committed:
Merge branch 'Aesthetic-top-and-bottom-surface-patterns' of https://github.com/pi-squared-studio/OrcaSlicer into Aesthetic-top-and-bottom-surface-patterns

pi-squared-studio committed:
Update strength_settings_infill_rotation_template_metalanguage.md

`α%` - that is coorect description
`α:β`, `ℍ:ℕ` - new instructions and its description

pi-squared-studio committed:
Merge branch 'main' into Aesthetic-top-and-bottom-surface-patterns

pi-squared-studio committed:
Update Field.cpp

pi-squared-studio committed:
Final release

pi-squared-studio committed:
Merge branch 'Aesthetic-top-and-bottom-surface-patterns' of https://github.com/pi-squared-studio/OrcaSlicer into Aesthetic-top-and-bottom-surface-patterns

pi-squared-studio committed:
Bug remove: surfaces with variable density

pi-squared-studio committed:
Precision Infill - Update & Check

pi-squared-studio commented:
Aesthetic Royal Flush:

<img width="960" height="450" alt="image" src="https://github.com/user-attachments/assets/dec1c493-48ad-45f2-99e7-6078a8c4b662" />
<img width="960" height="450" alt="image" src="https://github.com/user-attachments/assets/10d70918-48c9-47e4-8f77-9292382671b0" />

After calibrating all the parameters, I printed the credit card holder using all the advantages of this pack:
- Patchwork 20x20
- Combined Fuzzy Skin at walls
- Anisotropic surfaces
- Odd-Even perimeters

pi-squared-studio committed:
Update .gitignore

pi-squared-studio committed:
Merge branch 'main' into Aesthetic-top-and-bottom-surface-patterns

pi-squared-studio commented:
@SoftFever 
I would like to remind you once again that this project has been completed and is ready to be merged. 
Another moment - when I was working on it, I found a bug that infill/wall overlap in Classic mode calculates incorrectly. I was fix it

pi-squared-studio commented:
@ianalexis @SoftFever  I'm watching this video. https://www.youtube.com/watch?v=BCmSUQHVD6M it says about the rotating infill - there is an incorrectly described instruction in the Wiki that may lead to discrepancies.

<img width="1377" height="165" alt="image" src="https://github.com/user-attachments/assets/b1184a81-9661-4f54-b2b6-f3fe3b82c407" />

In this PR, I not only made the correct definition, but also added another one. Please adjust it.

ianalexis commented on doc/print_settings/strength/strength_settings_infill_rotation_template_metalanguage.md:
```diff
@@ -64,15 +68,15 @@ This metalanguage provides a way to define the [direction and rotation](strength
 - `|` - infill of multiple layers with vertical displacement at middle angle. e.g. `+22.5|50%`  
   ![multiple-joint-middle-angle](https://github.com/SoftFever/OrcaSlicer/blob/main/doc/images/fill/Template-metalanguage/multiple-joint-middle-angle.png?raw=true)
-- `N` - infill formed by sinus function (vertical connection). e.g. `+22.5N50%`  
+- `N` - infill formed by sinus function (vertical connection). e.g. `+22.5N50%`<br>If we reduce the angle and height by two, we get the same rotation of the infill, but with a greater frequency of waves.<br>
```
```suggestion
- `N` - infill formed by sinus function (vertical connection). e.g. `+22.5N50%`.  
  Reducing the angle and height by two gives the same rotation of the infill, the rotation of the infill remains unchanged, but the frequency of waves increases.  
```

ianalexis reviewed (changes requested):
I changed:
 - The HTML line break to the MD standard (which is a double space before a line break).
 - Some words and sentences that I think were errors in the translation.

Please review my two questions about interpretation in others_settings_special_mode.md, as I am not sure I interpreted the meaning of the sentence correctly.

pi-squared-studio committed:
Update doc/print_settings/strength/strength_settings_infill_rotation_template_metalanguage.md

Co-authored-by: Ian Bassi <ian.bassi@outlook.com>

pi-squared-studio committed:
Update doc/print_settings/strength/strength_settings_infill_rotation_template_metalanguage.md

Co-authored-by: Ian Bassi <ian.bassi@outlook.com>

pi-squared-studio committed:
Update doc/print_settings/strength/strength_settings_infill_rotation_template_metalanguage.md

Co-authored-by: Ian Bassi <ian.bassi@outlook.com>

pi-squared-studio committed:
Update doc/print_settings/strength/strength_settings_infill_rotation_template_metalanguage.md

Co-authored-by: Ian Bassi <ian.bassi@outlook.com>

pi-squared-studio committed:
Update doc/print_settings/strength/strength_settings_infill_rotation_template_metalanguage.md

Co-authored-by: Ian Bassi <ian.bassi@outlook.com>

pi-squared-studio committed:
Update doc/print_settings/strength/strength_settings_infill_rotation_template_metalanguage.md

Co-authored-by: Ian Bassi <ian.bassi@outlook.com>

afmenez commented on src/libslic3r/PrintConfig.cpp:
```diff
@@ -5570,6 +5588,145 @@ void PrintConfigDef::init_fff_params()
+    def           = this->add("anisotropic_surfaces", coBool);
+    def->label    = L("Anisotropic surfaces");
+    def->category = L("Strength");
+    def->tooltip  = L("Anisotropic patterns on the top and bottom surfaces."
```
Missing spaces after periods on the first two lines.

pi-squared-studio committed:
Merge branch 'SoftFever:main' into Aesthetic-top-and-bottom-surface-patterns

pi-squared-studio committed:
Wiki updates

pi-squared-studio committed:
Merge branch 'Aesthetic-top-and-bottom-surface-patterns' of https://github.com/pi-squared-studio/OrcaSlicer into Aesthetic-top-and-bottom-surface-patterns

pi-squared-studio committed:
Update wiki

pi-squared-studio committed:
+wiki

pi-squared-studio committed:
also wiki

pi-squared-studio committed:
wiki again

ianalexis commented:
I think you should rework this PR in the following way

- Anisotropic surfaces (Already done and merged so nothing to do here)
- Center of surface patterns (Already done and merged so nothing to do here)
- Precision surfaces: Make its own PR, i like this PR and think is necessary.
- Patchwork surfaces: Make its own PR, idk about this one its really advanced but maybe in a single PR is easier to think about it.
- Fixed: If possible to add to the Precision surfaces PR would be nice.

As always thanks for your work Pi!
Believe me, I understand how frustrating it is to have PRs that you've spent so much time on just gathering dust.
But anyway, now the expert menu lets us make these kinds of additions.

pi-squared-studio commented:
@ianalexis 
Yes, it's a big problem when I start doing minor improvements, and it turns into a big project that I don't have time to complete.
Regarding precise surfaces, as it turned out, this PR idea reached a dead end. I think that after refining Fuzzy Skin, I will return to it with new working principles.

afmenez commented on src/libslic3r/PrintConfig.cpp:
Extra space at the end of last line.

afmenez commented on src/libslic3r/PrintConfig.cpp:
Extra space at the end of last line.
